GiniVision/Classes/Core/GiniImageDocument.swift

```diff
@@ -21,6 +21,7 @@ final public class GiniImageDocument: NSObject, GiniVisionDocument {
     public var previewImage: UIImage?
     public var isReviewable: Bool
     public var isImported: Bool
+    public var error: Error?
```

Comment:
I cannot totally agree with the semantics here. It's convenient to have the error with the document. But semantically, a document is just a document. If it has an error, it cannot be constructed.

I would **seriously** consider moving it somewhere else.

Reply:
This was the last option for me. I *really* don't like having and error inside of a document, since as you well say, it should not be part of a document because a document is a document.
But the problem was that in the `CameraViewControllerDelegate` I have to send the documents back along with theirs errors (If they had).
Firstly I tried with a `[GiniVisionDocument: GiniError]` dictionary but `GiniVisionDocument` does not and can not implement `Hashable`. Secondly I tried the easy and ugly approach, using a tuple but it is not supported by ObjC.
The other option that I ruled out was to have a Wrapper or something similar, but in my opinion it can be more confusing for those who use the Component API.

Reply:
Still not a good reason. This thing will come back to haunt you. It's an architectural error that you will probably never fix. Even when you decide to roll out a new major version, this thing will be so deeply integrated that you will not dare to touch it.

Reply:
Do you have a better idea then? Because I know that is an architectural error but I don't see a better approach.